core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala

```diff
@@ -128,14 +128,17 @@ extends TimeSeriesStore with StrictLogging {
                      flushSched: Scheduler,
                      cancelTask: Task[Unit]): CancelableFuture[Unit] = {
     val shard = getShardE(dataset, shardNum)
-    shard.isReadyForQuery = true
     logger.info(s"Shard now ready for query dataset=$dataset shard=$shardNum")
     shard.shardStats.shardTotalRecoveryTime.update(System.currentTimeMillis() - shard.creationTime)
     shard.startIngestion(stream, cancelTask, flushSched)
   }
 
-  def recoverIndex(dataset: DatasetRef, shard: Int): Future[Long] =
-    getShardE(dataset, shard).recoverIndex()
+  def recoverIndex(dataset: DatasetRef, shard: Int): Future[Long] = {
+    val shard = getShardE(dataset, shard)
+    val fut = shard.recoverIndex()
+    fut.onComplete(_ => shard.isReadyForQuery = true)
```

Comment:
This is currently set in ``startIngestion`` in ``TimeSeriesMemStore`` and for non downsample this is the flow


```
for {
        _ <- tsStore.recoverIndex(ref, shard)
        checkpoints <- tsStore.metastore.readCheckpoints(ref, shard)
      } yield {
        if (checkpoints.isEmpty) {
          normalIngestion(...)
        } else {
          // Figure out recovery end watermark and intervals.  The reportingInterval is the interval at which
          // offsets come back from the MemStore for us to report progress.
           ...
          for {lastOffset <- doRecovery(...)}
            yield {
              normalIngestion(...)
            }
        }
```

This ``normalIngestion`` calls  ``startIngestion`` in ``TimeSeriesMemStore`` and that's where this flag is set to ``true``. By setting this flag from ``onComplete`` of recover we might be setting this bit too early.

CC @vishramachandran what's you take?